pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -281,16 +335,8 @@ def allocate_budget(
         else:
             constraints = custom_constraints
 
-        num_channels = len(self.parameters["channels"])
-        initial_guess = np.ones(num_channels) * total_budget / num_channels
-        bounds = [
-            (
-                (budget_bounds[channel][0], budget_bounds[channel][1])
-                if channel in budget_bounds
-                else (0, total_budget)
-            )
-            for channel in self.parameters["channels"]
-        ]
+        budgets_size = np.prod(self._budget_shape)
+        initial_guess = np.ones(budgets_size) * total_budget / budgets_size
```

Comment:
TODO: Re-implement old initial_guess based on bounds? Hard to make it work for higher-dimensional budgets

Reply:
The examples we have seem happy with this simple initial_guess

Reply:
seems areasonable